### PR TITLE
Improved Product pages templates management

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Templating of Product Pages are not really manageables, this is an attempt to improve the way templates are used and overriden. This is mainly about re-order and extract sub templates and blocks to make the customization and the maintenance simpler.
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | (optional) If this PR fixes a [Forge](http://forge.prestashop.com/) ticket, please add its complete Forge URL.
| How to test?  | The product pages should works as before

> I'll try to not make functional BC with `1.7.3.x` branch, this will only concerns what I call "Modern modules", the new feature of PrestaShop 1.7.3.

### Before

productpages

### After

carbon 5

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8489)

### CODE REVIEW

File: src/PrestaShopBundle/Resources/views/Admin/Product/CatalogPage/catalog.html.twig [hunk 1]
```diff
@@ -133,61 +133,45 @@
...
-                      'column': 'name'
-                    } %}
+                    {{ ps.carrets('name') }}
                   </th>
                   <th class="small-text" style="width: 9%">
                     {{ "Reference"|trans({}, 'Admin.Global') }}
-                    {% include '@Product/CatalogPage/catalog_order_carrets.html.twig' with {
-                      'column': 'reference'
-                    } %}
+                    {{ ps.carrets('reference') }}
```
Comment (kpodemski) [inline src/PrestaShopBundle/Resources/views/Admin/Product/CatalogPage/catalog.html.twig]: what is "carret" in this context? what does it mean?

Comment (mickaelandrieu) [reply src/PrestaShopBundle/Resources/views/Admin/Product/CatalogPage/catalog.html.twig]: hmm its not even english xD, renamed to sortArrows, thanks.

**Comment by mickaelandrieu** [comment]
@kpodemski what do you think? is it enough for module developers needs? /c @firstred

**Comment by prestarocket** [comment]
With your PR, is it possible to add a column in products list just with a hook ? (like hookActionAdminProductsListingFieldsModifier)

**Comment by mickaelandrieu** [comment]
@prestarocket with my PR you don't need a hook to do that, you only have to override the template `products_table.html.twig` :+1: 

See https://gist.github.com/mickaelandrieu/4a02ea05bb1113718aec5432ad418ee5

**Comment by prestarocket** [comment]
What happens if 2 modules override the same file ?

**Comment by mickaelandrieu** [comment]
> What happens if 2 modules override the same file ?

The latest template in loading will win. Do we need specific hook(s) to manage column(s) and if "yes" what is the best implementation? As you know, hooks cost a lot in performance and we try to avoid them (for now).

**Comment by LittleBigDev** [comment]
:lgtm:

---

Reviewed 47 of 56 files at r1, 2 of 3 files at r2, 9 of 16 files at r3, 1 of 4 files at r5, 14 of 14 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8489#-:-KzO5wp5oUPusy3PXBzk:bnfp4nl)*

**Comment by Quetzacoalt91** [comment]
@mickaelandrieu, it seems we need a rebase here after #8530

**Comment by mickaelandrieu** [comment]
@Quetzacoalt91 it's rebased !

**Comment by LittleBigDev** [comment]
Reviewed 1 of 56 files at r1, 5 of 5 files at r7.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8489)*

**Comment by mickaelandrieu** [comment]
@Quetzacoalt91 can you please review this one? I'd like to see it verified by QA asap

**Review by Quetzacoalt91 — DISMISSED** [review]
I'm cool with it, QA team can proceed. :)

**Comment by LittleBigDev** [comment]
Reviewed 1 of 56 files at r1, 2 of 2 files at r8.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8489)*

**Comment by marionf** [comment]
Hello @mickaelandrieu 

The image editing block is empty
capture du 2017-12-14 15-14-06

Labels are missing for delivery options
capture du 2017-12-14 15-30-42

When I check a supplier I can't save 
capture du 2017-12-14 15-42-44

And the form to add references isn't displayed (this one):
capture du 2017-12-14 15-39-23
I can see it if I refresh the page

**Comment by mickaelandrieu** [comment]
@marionf 

> The image editing block is empty

The error is also present in develop branch => not related 

> Labels are missing for delivery options

The error is also present in develop branch => not related 

> When I check a supplier I can't save And the form to add references isn't displayed (this one)

~~True story, I'll fix theses issues.~~ I've rebased from develop, these last 2 issues are fixed, you may need to install assets for both "default" and "new-theme" themes

**Review by Quetzacoalt91 — APPROVED** [review]

**Pull request merged**